File: features/support/hooks.rb

STRESS_TIMEOUT = 300


Before do |scenario|

  # fetch scenario and feature name, so we can use it in log files if needed
  case scenario
    when Cucumber::RunningTestCase::Scenario
      @feature_name = scenario.feature.name
      @scenario_title = scenario.name
    when Cucumber::RunningTestCase::ExampleRow
      @feature_name = scenario.scenario_outline.feature.name
      @scenario_title = scenario.scenario_outline.name
  end
  
  @load_method  = DEFAULT_LOAD_METHOD
  @query_params = {}
  @scenario_time = Time.now.strftime("%Y-%m-%dT%H:%m:%SZ")
  reset_data
  @has_logged_preprocess_info = false
  @has_logged_scenario_info = false
  set_grid_size DEFAULT_GRID_SIZE
  set_origin DEFAULT_ORIGIN

end

Around('@stress') do |scenario, block|
  Timeout.timeout(STRESS_TIMEOUT) do
    block.call
  end
end

After do
end

File: features/step_definitions/hooks.js
var util = require('util');

module.exports = function () {
    this.Before((scenario, callback) => {
        this.scenarioTitle = scenario.getName();

        this.loadMethod = this.DEFAULT_LOAD_METHOD;
        this.queryParams = {};
        var d = new Date();
        this.scenarioTime = util.format('%d-%d-%dT%s:%s:%sZ', d.getFullYear(), d.getMonth()+1, d.getDate(), d.getHours(), d.getMinutes(), d.getSeconds());
        this.resetData();
        this.hasLoggedPreprocessInfo = false;
        this.hasLoggedScenarioInfo = false;
        this.setGridSize(this.DEFAULT_GRID_SIZE);
        this.setOrigin(this.DEFAULT_ORIGIN);
        callback();
    });
};
